File: src/encode.ts
export function selectEncoding(encoding: string | string[]) {
    if (!encoding) return null
    if (typeof encoding == 'string') encoding = [encoding]

    const encodings = encoding.map(e => e.toLowerCase())
    if (encodings.includes('gzip')) return 'gzip'
    // if (encodings.includes('deflate')) return 'deflate'
    // if (encodings.includes('br')) return 'br'
    // if (encodings.includes('identity')) return 'identity'
    
    return null
}

File: src/encode.cpp
#include "encode.hpp"

#include <string>
#include <vector>
#include <algorithm>

#include "utils/console.hpp"
#include "utils/utils.hpp"

using std::string;
using std::vector;

string selectEncoding(vector<string> encodings) {
    // Supported encodings
    if (std::find(encodings.begin(), encodings.end(), "gzip") != encodings.end()) return "gzip";

    return "";
}